src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx

```diff
@@ -98,6 +99,10 @@ const calculateSpendingLimitsTxData = (
     transactions.push(addSpendingLimitBeneficiaryMultiSendTx(values.beneficiary))
   }
 
+  if (existentSpendingLimit && existentSpendingLimit.spent !== '0') {
+    transactions.push(resetSpendingLimitBeneficiaryMultiSendTx(existentSpendingLimit.delegate, txToken.address))
```

Comment:
Do I get this right that once I make a transaction within my limit, I'll be able to make another transaction within the same limit?
For example, the limit is 10 ETH. I send 8 ETH. It automatically resets. I can send 8 ETH again.

And before this change, if I send 8 ETH, I can only send 2 ETH afterwards?

Reply:
Yes, that's how i understand it should work.

Reply:
I though @katspaugh was saying that. @nicosampler, what you are saying is correct.

Reply:
Oh, OK. Thanks Nico! This behavior makes a lot of sense.